Add Tool Info for JDart

mmuesly opened the pull request:
We want to integrate JDart into the BenchExec framework.
Currently, JDart doesn't support a fixed Version system that is extractable from the binary.  Therefore, I left the Version String empty.

PhilippWendler commented:
Do you want to participate in Test-Comp or SV-COMP? Note that for these competitions it is required to have a version in the tool that is extracted by this tool-info module.

mmuesly commented:
Yes, we are planing to participate in SV-COMP with this tools.

It would be nice, if this requirement might be documented in the [rule section](https://sv-comp.sosy-lab.org/2020/rules.php) or somewhere else in the future. At the moment, it is written in various places that returning a version is desired, but optional.

I will rework the PR once the tool is adapted.

mmuesly commented:
I updated the PR and made the artifact able to provide a version on request.

Earlier in the day, I had a discussion with Dirk Beyer on JVM provisioning, in case I want to use a non-default JVM.
At the end, I decided to include the JVM version that is suitable in the artifact. After reading up on the discussion in #453, my understanding is that the REQUIRED_PATHS variable needs to list all files on the top-level explicitly and every subfolder that should be included completely in the  execution environment. I therefore adapted the REQUIRED_PATHS to match our artifact with this interpretation of the code.
Could you shortly confirm this assumption? Is there an easy way to test that all files end up at the right place in this distribution step?

Thanks!

PhilippWendler commented:
Your assumption is correct. You can test it using [these instructions](https://github.com/sosy-lab/benchexec/blob/master/doc/tool-integration.md#testing-the-tool-integration), it will output the paths that it will use.

In the distribution step, everything listed as part of these paths is copied to a different location, so make sure that you do not use any absolute paths anywhere, but the relative location of the paths will stay the same.

PhilippWendler commented:
> It would be nice, if this requirement might be documented in the [rule section](https://sv-comp.sosy-lab.org/2020/rules.php) or somewhere else in the future. At the moment, it is written in various places that returning a version is desired, but optional.

@dbeyer

dbeyer commented:
Thanks for the hint about the version reporting.  I thought every tool has this anyway. Now added to the rules page.

dbeyer commented:
@PhilippWendler Is this ready to be merged?

PhilippWendler commented:
Yes, though all paths below `jdk8u222-b10/...` in `REQUIRED_PATHS` are redundant, and I think @mmuesly wanted to test it first.

mmuesly commented:
I tested the tool integration as described. In the program files output section, the expanded path for 'jdk8u222-b10' is listed but not all subfiles.

I trust @PhilippWendler that this is sufficient as he knows more about the tool than I do. Hence, I simplified the ´REQUIRED:PATHS´and squashed the commits. From my point of view, this should be ready to merge.

Thank you for the comments and support!

PhilippWendler commented:
If a directory is listed, everything in it is copied, yes.
